charmcraft/commands/store/__init__.py

```diff
@@ -246,41 +246,67 @@ class WhoamiCommand(BaseCommand):
     """
     )
 
+    def fill_parser(self, parser):
+        """Add own parameters to the general parser."""
+        self.include_format_option(parser)
+
     def run(self, parsed_args):
         """Run the command."""
         store = Store(self.config.charmhub)
         try:
             macaroon_info = store.whoami()
         except CredentialsUnavailable:
-            emit.message("You are not logged in to Charmhub.")
+            if parsed_args.format:
+                info = {"logged": False}
+                emit.message(self.format_content(parsed_args.format, info))
+            else:
+                emit.message("You are not logged in to Charmhub.")
             return
 
-        emit.message(f"name: {macaroon_info.account.name}")
-        emit.message(f"username: {macaroon_info.account.username}")
-        emit.message(f"id: {macaroon_info.account.id}")
+        human_msgs = []
+        prog_info = {"logged": True}
+
+        human_msgs.append(f"name: {macaroon_info.account.name}")
+        prog_info["name"] = macaroon_info.account.name
+        human_msgs.append(f"username: {macaroon_info.account.username}")
+        prog_info["username"] = macaroon_info.account.username
+        human_msgs.append(f"id: {macaroon_info.account.id}")
+        prog_info["id"] = macaroon_info.account.id
```

Comment:
The spec consider the first three attributes as mandatory. That is not useful when not logged, so I made them optional and added a `logged` attribute that the developer can rely on.